src/main/java/uk/ac/ebi/eva/evaseqcol/service/SeqColLevelTwoService.java

```diff
@@ -35,59 +36,58 @@ public Optional<SeqColLevelTwoEntity> getSeqColLevelTwoByDigest(String digest) {
             return Optional.empty();
         }
         // 2 DATABASE LOOKUPS
-        List<SeqColExtendedDataEntity> extendedAttributes = getExtendedAttributes(levelOneEntity.get());
+        List<SeqColExtendedDataEntity<?>> extendedIntegerAttributes = getIntegerExtendedAttributes(levelOneEntity.get());
+        List<SeqColExtendedDataEntity<?>> extendedStringAttributes = getStringExtendedAttributes(levelOneEntity.get());
         SeqColLevelTwoEntity levelTwoEntity = new SeqColLevelTwoEntity();
-        for (SeqColExtendedDataEntity extendedData: extendedAttributes) {
+        for (SeqColExtendedDataEntity<?> extendedData: extendedStringAttributes) {
             switch (extendedData.getAttributeType()) {
-                case lengths:
-                    levelTwoEntity.setLengths(extendedData.getExtendedSeqColData().getObject());
-                    break;
                 case names:
-                    levelTwoEntity.setNames(extendedData.getExtendedSeqColData().getObject());
+                    levelTwoEntity.setNames((List<String>) extendedData.getExtendedSeqColData().getObject());
```

Comment:
I am surprised that this kind of casting works without explicitly invoking toString on every element!